base/mp_getddl.sas

```diff
@@ -172,28 +176,16 @@ run;
       put name fmt notnul;
     run;
     data _null_;
-      length ds $128;
-      set &idxinfo (where=(memname="&curds"));
+      length ctype $11;
+      set &tabconst (where=(table_name="&curds")) end=last;
       file &fref mod;
-      by idxusage indxname;
-      if unique='yes' then uniq=' unique';
-      ds=cats(libname,'.',memname);
-      if first.indxname then do;
-        if unique='yes' and nomiss='yes' then do;
-          put '  ,constraint [' indxname '] PRIMARY KEY';
-        end;
-        else if unique='yes' then do;
-          /* add nonclustered in case of multiple unique indexes */
-          put '  ,index [' indxname '] UNIQUE NONCLUSTERED';
-        end;
-        put '  (';
-        put '    [' name ']';
-      end;
-      else put '    ,[' name ']';
-      if last.indxname then do;
-        put '  )';
-      end;
+      by constraint_type constraint_name;
+      if upcase(strip(constraint_type)) = 'PRIMARY' then ctype='PRIMARY KEY';
+      else ctype=strip(constraint_type);
+      column_list=strip(column_list);
+      put "   ,CONSTRAINT " constraint_name " " ctype "(" column_list ")";
```

Comment:
the format of this constraint does not reflect the logic it replaces. Note the square brackets - these are necessary for running TSQL flavoured DDL.  Also the syntax needs to remain the same as it was before.

finally the "column_list" approach is risky.